extension/js/common/core/crypto/key.ts

```diff
@@ -125,6 +125,10 @@ export class KeyUtil {
   }
 
   public static parse = async (text: string): Promise<Key> => {
+    const keyLength = (await KeyUtil.parseMany(text)).length;
+    if (keyLength > 1) {
+      throw new Error(`There are ${keyLength} keys found, expecting 1 x509 key`);
```

Comment:
This method is used for both x509 and OpenPGP keys, they are not differentiated at this level.
This code here is probably not required as an exception will come up from underlying Smimekey or OpenPGP parsers.

Reply:
Hi @rrrooommmaaa,

Thank you very much for the helpful details. I'm still not sure If I'm doing things correctly, sorry about that.

It tried to `const pubKey = [testConstants.testkey715EDCDC7939A8F7, testConstants.pubkey2864E326A5BE488A].join('');` on my test and pass it to `await KeyUtil.parse(pubKey)` and this always return an object which I suspect the came from `(await KeyUtil.parseMany(text))[0]`.

I think of counting the `allIds` property but I'm still unsure, what exactly needs to be count when we talk about `There are 2 OpenPGP keys found, expected just one`?

I also remove all and reverted back the changes from the previous commit.

Can I get another hint, please? Sorry for the inconvenience.

Reply:
Hi @martgil. Please commit the unit test, so I can run it and see what is missing.
Yes, joining these 2 keys would work (only I guess we should join them with \r\n)
`const keys = [testConstants.testkey715EDCDC7939A8F7, testConstants.pubkey2864E326A5BE488A].join('\r\n');`
in the test, call `KeyUtil.parseMany(keys)` and make sure it returns an array of length === 2.
and then test that KeyUtil.parse(keys) throws an exception as I previosuly mentioned.
and test that OpenPGPKey.parse(keys) throws an exception.
Looks like we need this piece of code in KeyUtil.parse to throw the exception, but without specifying the type of the key (as we don't know the type of the keys at this level).

Reply:
Thanks @rrrooommmaaa,

I will. Unfortunately the `KeyUtil.parseMany(keys)` and `OpenPGPKey.parseMany(keys)` still return an object with length `1` from `const keys = [testConstants.testkey715EDCDC7939A8F7,testConstants.pubkey2864E326A5BE488A].join('\r\n');`. 

>Looks like we need this piece of code in KeyUtil.parse to throw the exception, but without specifying the type of the key

I have already checked and confirmed that those two `const keys` are `openpgp` type when I look into the `.type` property.

Reply:
That's right. Looks like KeyUtil.parseMany() really doesn't support reading multiple keys from text.
when calling openPGP's readArmored(), for example, parsing stops after reading the first packet (private or public key).
So, do we need to fix KeyUtil.parseMany() first, @tomholub?
If so, it should be refactored to use `MsgBlockParser.detectBlocks()` instead of `KeyUtil.getKeyType()` switch?
Should there be errors if we have mixed content -- OpenPGP('privateKey', 'publicKey') and S/MIME (others)?

Reply:
It turned out 2 keys should be wrapped under one armor. I fixed the test.

Reply:
Thank you sir @rrrooommmaaa, I have a look into these latest changes on my local machine. Now I understand that the two OpenPGP keys should be "dearmor" and "armor" and somewhat like "repack" all together into a single PGP key block. Btw, are the term "armor" and "deamor" refers to a public key without comment and headers and vise-versa?

Thanks Again!
Mart Gil

Reply:
Hi @martgil
Any type of data can be wrapped in armor (Base64 encoding + delimiters (possibly a comment line too)).
In this example, the parameter `opgp.enums.armor.public_key` will lead to the header `-----BEGIN PGP PUBLIC KEY BLOCK-----`. etc.
Also, MessageBlockParser.detectBlocks determines input type (including S/MIME types), etc.
`dearmor` function returns a structure with two fields: `type`  and ` data` -- the parameters used to `armor`